### ISSUE TITLE
Internal Routes for a Route-Less Engine

### ISSUE THREAD

**Issue by anitricks** [ember-engines/ember-engines#624]
I need help with a concept:

I need to develop a chat-box that has its own internal routes like - /chat/:chat_id, /chat/:type - and mount it into an ember application. However, the chat-box is an overlay in the application. By that I mean the chat-box doesn't disturb the main ember application routes. For example, moving from route-to-route in the ember app doesn't change the engines routes and vice-versa.

Sorry, I am new to ember and its concept so this might be a stupid question. I have read the docs for ember-engines already but was wondering if internal routing is possible for an engine. I know the router is shared but is there a way for an engine to have its own route

**Comment by villander** [comment]
yup, it's possible...please check the Ember Engines Guides on section [Adding Routes for Routable Engines](http://ember-engines.com/guide/creating-an-engine#AddingRoutesforRoutableEngines)

**Comment by bertdeblock** [comment]
If I understand correctly, the question is if a *route-less* engine can have routes internally, which I don't think is possible?
